src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String SUPPLY_LABEL = MarketItemType.SUPPLY.toString().toLowerCase();
+    private static final String BUY_LABEL = MarketItemType.BUY.toString().toLowerCase();
+    private static final String RESULT_LABEL = "result";
+    private static final String COMMA = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    public String getStatistic(String fromFileName, String toFileName) {
+        MarketData marketData = new MarketData();
+
+        readFromFile(fromFileName, (line) -> {
+            String[] tokens = line.split(COMMA);
+            if (line.isEmpty() || tokens.length != 2) {
+                return;
+            }
+            String type = tokens[0].toLowerCase().trim();
+            int value = Integer.parseInt(tokens[1].trim());
+            if (type.equals(SUPPLY_LABEL)) {
+                marketData.addSupplyTotal(value);
+            } else if (type.equals(BUY_LABEL)) {
+                marketData.addBuyTotal(value);
+            }
```

Unknown operation types are currently ignored silently. The requirements demand that this behavior be explicit. Either throw a RuntimeException for an unknown operation or explicitly ignore it with a clear comment. Quoted requirement: "Handle unknown operation types by either ignoring them explicitly or throwing a RuntimeException; behavior must be defined in code and not silently fail." Make your chosen behavior explicit in code and provide a clear error message if you throw.